src/server/game/Entities/Player/Player.cpp

```diff
@@ -15852,6 +15858,17 @@ void Player::RemoveRewardedQuest(uint32 questId, bool update /*= true*/)
         m_RewardedQuestsSave[questId] = QUEST_FORCE_DELETE_SAVE_TYPE;
     }
 
+    // Remove seasonal quest also
+    if (Quest const* qInfo = sObjectMgr->GetQuestTemplate(questId))
+    {
+        uint16 eventId = sGameEventMgr->GetEventIdForQuest(qInfo);
```

Comment:
D'oh. You removed `GetEventIdForQuest()` yourself in your PR https://github.com/TrinityCore/TrinityCore/pull/18684 in December.